### DISCUSSION TITLE
[Feature proposal] Locking up db schema.

### DISCUSSION THREAD

**Discussion by KonradHEXANT** [directus/directus#3805] [Feature Requests]
During a couple of previous projects that involved "CMS that can change DB structure" I was facing the problem of data/app corruption by "accidental clicks" from unaware users. Yes, I know that there are roles and permissions that can prevent that to some extent, but this problem is still valid if "unauthorized" access occurs.
What I want to suggest is adding an "env" flag like DB_STRUCTURE_LOCKED = true that should simply hide elements for data structure management once an app is delivered to a client, to avoid such accidents.

**Comment by philleepflorence** [comment]
Can this be solved by using or limiting the CORS Methods?
https://docs.directus.io/reference/environment-variables.html#cors

> This might affect Activity and Authentication...

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
